File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-14.cpp
using namespace std;
class Solution {
private:
    stack<int> numStack;

public:
    int calculate(int num1, int num2, string sign) {
        if (sign == "+") {
            return num1 + num2;
        }
        if (sign == "-") {
            return num1 - num2;
        }
        if (sign == "*") {
            return num1 * num2;
        }
        return num1 / num2;
    }

    bool isOperator(string token) {
        return (token == "+" or token == "-" or token == "*" or token == "/");
    }

    int evalRPN(vector<string>& tokens) {
        for (string token : tokens) {
            if (isOperator(token)) {
                int num1 = numStack.top();
                numStack.pop();
                int num2 = numStack.top();
                numStack.pop();
                numStack.push(calculate(num2, num1, token));
            } else {
                numStack.push(stoi(token));
            }
        }
        return numStack.top();
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-14.py
class Solution:
    def isValid(self, s: str) -> bool:
        opened = {')': '(', '}': '{', ']': '['}
        stack = []

        for bracket in s:
            if stack and bracket in opened:
                if stack[-1] == opened[bracket]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(bracket)
        
        return not stack